src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1025,28 +1025,101 @@ public INamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
         private readonly ConcurrentCache<string, INamedTypeSymbol?> _getTypeCache =
             new ConcurrentCache<string, INamedTypeSymbol?>(50, ReferenceEqualityComparer.Instance);
 
+        private readonly ConcurrentCache<string, ImmutableArray<INamedTypeSymbol>> _getTypesCache =
+            new ConcurrentCache<string, ImmutableArray<INamedTypeSymbol>>(50, ReferenceEqualityComparer.Instance);
+
         /// <summary>
         /// Gets the type within the compilation's assembly and all referenced assemblies (other than
         /// those that can only be referenced via an extern alias) using its canonical CLR metadata name.
-        /// </summary>
-        /// <returns>Null if the type can't be found.</returns>
+        /// This lookup follows the following order:
+        /// <list type="number">
+        /// <item><description>If the type is found in the compilation's assembly, that type is returned.</description></item>
+        /// <item><description>Next, the core library (the library that defines <c>System.Object</c>) is searched. If the type is found there, that type is returned.</description></item>
```

Comment:
>the core library [](http://example.com/codeflow?start=37&length=16)

It would be good to clarify if the core library would be searched if it is references via an extern alias.

Reply:
Clarified this in the remarks below.